File: server/models/product.go
package models

import "gorm.io/gorm"

type Category struct {
	gorm.Model
	ID           uint   `gorm:"primaryKey" json:"id"`
	CategoryName string `gorm:"size:50;not null;unique" json:"category_name"`
}

type Product struct {
	gorm.Model
	ID             uint        `gorm:"primaryKey;autoIncrement" json:"id"`
	ProductName    string      `gorm:"size:100;not null" json:"product_name"`
	CategoryID     uint        `gorm:"foreignKey:CategoryID;not null" json:"category_id"`
	MaxTemperature float64     `gorm:"type:decimal(10,2);" json:"max_temperature"`
	MinTemperature float64     `gorm:"type:decimal(10,2);" json:"min_temperature"`
	SpecWeight     float64     `gorm:"type:decimal(10,2);" json:"spec_weight"`
	SpecVolume     float64     `gorm:"type:decimal(10,2);" json:"spec_volume"`
	ImageURL       string      `gorm:"size:255;" json:"image_url"`
	OrderItems     []OrderItem `gorm:"foreignKey:ProductID" json:"order_items"`
	Category       Category    `gorm:"foreignKey:CategoryID" json:"category"`
}

File: mobile-web/src/interface/order/product.ts
export interface ProductType {
    id: number // 产品ID
    product_name: string // 产品名称
    category_name: string
    min_temperature: number // 最低温度 (°C)
    max_temperature: number // 最高温度 (°C)
    spec_weight: number // 产品重量 (kg)
    spec_volume: number // 体积 (m³)
    image_url: string // 产品图片
}
